### DISCUSSION TITLE
How to build OTD properly?

### DISCUSSION THREAD

**Discussion by JimmyCushnie** [OpenTabletDriver/OpenTabletDriver#1298] [Q&A] [answered]
Hi, I'm a bit of a noob but I'd like to play around with the OTD code and potentially contribute to development, starting with completing support for my device (adding tilt to Huion H1060p). However, I can't figure out how to build and run the project properly.

**What I'm doing:** Using Visual Studio to run "build solution" on the `OpenTabletDriver` solution, or on the `OpenTabletDriver.Desktop` project.

**What I expect to happen:** There's a `bin` folder somewhere that contains `OpenTabletDriver.Daemon.exe`, `OpenTabletDriver.UX.Wpf.exe`, and a `Configurations` folder -- same as I get when I download from the Releases page.

**What actually happens:** All the `bin` folders contain dlls, and there are waaaaay more generated files than I see in the download.

I'm on Windows, by the way. I notice that all the projects are set to build as "class library" rather than as an executable, so this seems to be the intended behavior. But then, how do I build the exes from the release downloads? How can I actually test my changes when I'm modifying the code?

Thanks in advance for any help, and thank you so much for this project. I'm so grateful that I can draw in peace without being constantly spied on by Huion.

**Comment by adryzz** [comment, marked as answer]
> All the bin folders contain dlls, and there are waaaaay more generated files than I see in the download.

The downloads have the DLLs packaged in them. It is normal that you will have all the DLLs not packaged.

The bin folder contains also the executables, if you check.

The run button on VS should work, but if it doesn't, open up a command prompt and type `dotnet run` in the folder of the project you wanna test run.

**Comment by david-polak** [comment]
@JimmyCushnie

* Set the `Output path` in properties-build for every project to be a shared folder `build/` somewhere.
* Copy Configurations from the `OpenTabletDriver` project into `build/net5/Configurations`
* Rebuild the solution
* `.Deamon` and `.UX.<pltf>` should now be run-able

I also modified the `TargetFramework` for `.Desktop` and `.Tools` to `net5` so they end up in the same folder, no idea if that's gonna cause any issues yet. I cloned the project about 15 minutes ago, so I don't know if that's the right way of doing it.
